### ISSUE TITLE
[eastl] Build error

### ISSUE THREAD

**Issue by bboysnick5** [microsoft/vcpkg#22964]
Hi vcpkg team,

I have tried a few times installing eastl but failed. Below are the logs. Please advise. 

**Host Environment**
- OS: [e.g. Windows/Linux etc...]
  Windows
- Compiler: revision
MSVC 19.31.31103.0
**To Reproduce**
`./vcpkg install eastl`

**Failure logs** 
-Cut and paste the appropriate build messages from the console output.
```
[1/12] C:\PROGRA~1\MIB055~1\2022\Preview\VC\Tools\MSVC\1431~1.311\bin\Hostx64\x64\cl.exe   /TP -DEASTL_OPENSOURCE=1 -D_CHAR16T -D_CRT_SECURE_NO_WARNINGS -D_SCL_SECURE_NO_WARNINGS -IC:\Users\Nick\vcpkg\buildtrees\eastl\src\5ab9f1222a-26ecaf23c0.clean\include -external:IC:\Users\Nick\vcpkg\installed\x64-windows\include -external:W0 /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /GR /EHsc /MP  /std:c++latest /W4 /permissive- /D_DEBUG /MDd /Z7 /Ob0 /Od /RTC1 /showIncludes /FoCMakeFiles\EASTL.dir\source\numeric_limits.cpp.obj /FdCMakeFiles\EASTL.dir\EASTL.pdb /FS -c C:\Users\Nick\vcpkg\buildtrees\eastl\src\5ab9f1222a-26ecaf23c0.clean\source\numeric_limits.cpp
cl : Command line warning D9025 : overriding '/W3' with '/W4'
[2/12] C:\PROGRA~1\MIB055~1\2022\Preview\VC\Tools\MSVC\1431~1.311\bin\Hostx64\x64\cl.exe   /TP -DEASTL_OPENSOURCE=1 -D_CHAR16T -D_CRT_SECURE_NO_WARNINGS -D_SCL_SECURE_NO_WARNINGS -IC:\Users\Nick\vcpkg\buildtrees\eastl\src\5ab9f1222a-26ecaf23c0.clean\include -external:IC:\Users\Nick\vcpkg\installed\x64-windows\include -external:W0 /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /GR /EHsc /MP  /std:c++latest /W4 /permissive- /D_DEBUG /MDd /Z7 /Ob0 /Od /RTC1 /showIncludes /FoCMakeFiles\EASTL.dir\source\allocator_eastl.cpp.obj /FdCMakeFiles\EASTL.dir\EASTL.pdb /FS -c C:\Users\Nick\vcpkg\buildtrees\eastl\src\5ab9f1222a-26ecaf23c0.clean\source\allocator_eastl.cpp
cl : Command line warning D9025 : overriding '/W3' with '/W4'
[3/12] C:\PROGRA~1\MIB055~1\2022\Preview\VC\Tools\MSVC\1431~1.311\bin\Hostx64\x64\cl.exe   /TP -DEASTL_OPENSOURCE=1 -D_CHAR16T -D_CRT_SECURE_NO_WARNINGS -D_SCL_SECURE_NO_WARNINGS -IC:\Users\Nick\vcpkg\buildtrees\eastl\src\5ab9f1222a-26ecaf23c0.clean\include -external:IC:\Users\Nick\vcpkg\installed\x64-windows\include -external:W0 /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /GR /EHsc /MP  /std:c++latest /W4 /permissive- /D_DEBUG /MDd /Z7 /Ob0 /Od /RTC1 /showIncludes /FoCMakeFiles\EASTL.dir\source\intrusive_list.cpp.obj /FdCMakeFiles\EASTL.dir\EASTL.pdb /FS -c C:\Users\Nick\vcpkg\buildtrees\eastl\src\5ab9f1222a-26ecaf23c0.clean\source\intrusive_list.cpp
cl : Command line warning D9025 : overriding '/W3' with '/W4'
[4/12] C:\PROGRA~1\MIB055~1\2022\Preview\VC\Tools\MSVC\1431~1.311\bin\Hostx64\x64\cl.exe   /TP -DEASTL_OPENSOURCE=1 -D_CHAR16T -D_CRT_SECURE_NO_WARNINGS -D_SCL_SECURE_NO_WARNINGS -IC:\Users\Nick\vcpkg\buildtrees\eastl\src\5ab9f1222a-26ecaf23c0.clean\include -external:IC:\Users\Nick\vcpkg\installed\x64-windows\include -external:W0 /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /GR /EHsc /MP  /std:c++latest /W4 /permissive- /D_DEBUG /MDd /Z7 /Ob0 /Od /RTC1 /showIncludes /FoCMakeFiles\EASTL.dir\source\fixed_pool.cpp.obj /FdCMakeFiles\EASTL.dir\EASTL.pdb /FS -c C:\Users\Nick\vcpkg\buildtrees\eastl\src\5ab9f1222a-26ecaf23c0.clean\source\fixed_pool.cpp
cl : Command line warning D9025 : overriding '/W3' with '/W4'
[5/12] C:\PROGRA~1\MIB055~1\2022\Preview\VC\Tools\MSVC\1431~1.311\bin\Hostx64\x64\cl.exe   /TP -DEASTL_OPENSOURCE=1 -D_CHAR16T -D_CRT_SECURE_NO_WARNINGS -D_SCL_SECURE_NO_WARNINGS -IC:\Users\Nick\vcpkg\buildtrees\eastl\src\5ab9f1222a-26ecaf23c0.clean\include -external:IC:\Users\Nick\vcpkg\installed\x64-windows\include -external:W0 /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /GR /EHsc /MP  /std:c++latest /W4 /permissive- /D_DEBUG /MDd /Z7 /Ob0 /Od /RTC1 /showIncludes /FoCMakeFiles\EASTL.dir\source\red_black_tree.cpp.obj /FdCMakeFiles\EASTL.dir\EASTL.pdb /FS -c C:\Users\Nick\vcpkg\buildtrees\eastl\src\5ab9f1222a-26ecaf23c0.clean\source\red_black_tree.cpp
cl : Command line warning D9025 : overriding '/W3' with '/W4'
[6/12] C:\PROGRA~1\MIB055~1\2022\Preview\VC\Tools\MSVC\1431~1.311\bin\Hostx64\x64\cl.exe   /TP -DEASTL_OPENSOURCE=1 -D_CHAR16T -D_CRT_SECURE_NO_WARNINGS -D_SCL_SECURE_NO_WARNINGS -IC:\Users\Nick\vcpkg\buildtrees\eastl\src\5ab9f1222a-26ecaf23c0.clean\include -external:IC:\Users\Nick\vcpkg\installed\x64-windows\include -external:W0 /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /GR /EHsc /MP  /std:c++latest /W4 /permissive- /D_DEBUG /MDd /Z7 /Ob0 /Od /RTC1 /showIncludes /FoCMakeFiles\EASTL.dir\source\hashtable.cpp.obj /FdCMakeFiles\EASTL.dir\EASTL.pdb /FS -c C:\Users\Nick\vcpkg\buildtrees\eastl\src\5ab9f1222a-26ecaf23c0.clean\source\hashtable.cpp
cl : Command line warning D9025 : overriding '/W3' with '/W4'
[7/12] C:\PROGRA~1\MIB055~1\2022\Preview\VC\Tools\MSVC\1431~1.311\bin\Hostx64\x64\cl.exe   /TP -DEASTL_OPENSOURCE=1 -D_CHAR16T -D_CRT_SECURE_NO_WARNINGS -D_SCL_SECURE_NO_WARNINGS -IC:\Users\Nick\vcpkg\buildtrees\eastl\src\5ab9f1222a-26ecaf23c0.clean\include -external:IC:\Users\Nick\vcpkg\installed\x64-windows\include -external:W0 /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /GR /EHsc /MP  /std:c++latest /W4 /permissive- /D_DEBUG /MDd /Z7 /Ob0 /Od /RTC1 /showIncludes /FoCMakeFiles\EASTL.dir\source\string.cpp.obj /FdCMakeFiles\EASTL.dir\EASTL.pdb /FS -c C:\Users\Nick\vcpkg\buildtrees\eastl\src\5ab9f1222a-26ecaf23c0.clean\source\string.cpp
cl : Command line warning D9025 : overriding '/W3' with '/W4'
[8/12] C:\PROGRA~1\MIB055~1\2022\Preview\VC\Tools\MSVC\1431~1.311\bin\Hostx64\x64\cl.exe   /TP -DEASTL_OPENSOURCE=1 -D_CHAR16T -D_CRT_SECURE_NO_WARNINGS -D_SCL_SECURE_NO_WARNINGS -IC:\Users\Nick\vcpkg\buildtrees\eastl\src\5ab9f1222a-26ecaf23c0.clean\include -external:IC:\Users\Nick\vcpkg\installed\x64-windows\include -external:W0 /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /GR /EHsc /MP  /std:c++latest /W4 /permissive- /D_DEBUG /MDd /Z7 /Ob0 /Od /RTC1 /showIncludes /FoCMakeFiles\EASTL.dir\source\assert.cpp.obj /FdCMakeFiles\EASTL.dir\EASTL.pdb /FS -c C:\Users\Nick\vcpkg\buildtrees\eastl\src\5ab9f1222a-26ecaf23c0.clean\source\assert.cpp
cl : Command line warning D9025 : overriding '/W3' with '/W4'
[9/12] C:\PROGRA~1\MIB055~1\2022\Preview\VC\Tools\MSVC\1431~1.311\bin\Hostx64\x64\cl.exe   /TP -DEASTL_OPENSOURCE=1 -D_CHAR16T -D_CRT_SECURE_NO_WARNINGS -D_SCL_SECURE_NO_WARNINGS -IC:\Users\Nick\vcpkg\buildtrees\eastl\src\5ab9f1222a-26ecaf23c0.clean\include -external:IC:\Users\Nick\vcpkg\installed\x64-windows\include -external:W0 /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /GR /EHsc /MP  /std:c++latest /W4 /permissive- /D_DEBUG /MDd /Z7 /Ob0 /Od /RTC1 /showIncludes /FoCMakeFiles\EASTL.dir\source\atomic.cpp.obj /FdCMakeFiles\EASTL.dir\EASTL.pdb /FS -c C:\Users\Nick\vcpkg\buildtrees\eastl\src\5ab9f1222a-26ecaf23c0.clean\source\atomic.cpp
FAILED: CMakeFiles/EASTL.dir/source/atomic.cpp.obj 
C:\PROGRA~1\MIB055~1\2022\Preview\VC\Tools\MSVC\1431~1.311\bin\Hostx64\x64\cl.exe   /TP -DEASTL_OPENSOURCE=1 -D_CHAR16T -D_CRT_SECURE_NO_WARNINGS -D_SCL_SECURE_NO_WARNINGS -IC:\Users\Nick\vcpkg\buildtrees\eastl\src\5ab9f1222a-26ecaf23c0.clean\include -external:IC:\Users\Nick\vcpkg\installed\x64-windows\include -external:W0 /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /GR /EHsc /MP  /std:c++latest /W4 /permissive- /D_DEBUG /MDd /Z7 /Ob0 /Od /RTC1 /showIncludes /FoCMakeFiles\EASTL.dir\source\atomic.cpp.obj /FdCMakeFiles\EASTL.dir\EASTL.pdb /FS -c C:\Users\Nick\vcpkg\buildtrees\eastl\src\5ab9f1222a-26ecaf23c0.clean\source\atomic.cpp
cl : Command line warning D9025 : overriding '/W3' with '/W4'
C:\Users\Nick\vcpkg\buildtrees\eastl\src\5ab9f1222a-26ecaf23c0.clean\include\EASTL\internal\atomic\atomic_integral.h(330): error C2338: eastl::atomic<T> atomic macro not implemented!
C:\Users\Nick\vcpkg\buildtrees\eastl\src\5ab9f1222a-26ecaf23c0.clean\include\EASTL\internal\atomic\atomic_integral.h(330): note: see reference to class template instantiation 'eastl::internal::atomic_integral_width<T,16>' being compiled
[10/12] C:\PROGRA~1\MIB055~1\2022\Preview\VC\Tools\MSVC\1431~1.311\bin\Hostx64\x64\cl.exe   /TP -DEASTL_OPENSOURCE=1 -D_CHAR16T -D_CRT_SECURE_NO_WARNINGS -D_SCL_SECURE_NO_WARNINGS -IC:\Users\Nick\vcpkg\buildtrees\eastl\src\5ab9f1222a-26ecaf23c0.clean\include -external:IC:\Users\Nick\vcpkg\installed\x64-windows\include -external:W0 /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /GR /EHsc /MP  /std:c++latest /W4 /permissive- /D_DEBUG /MDd /Z7 /Ob0 /Od /RTC1 /showIncludes /FoCMakeFiles\EASTL.dir\source\thread_support.cpp.obj /FdCMakeFiles\EASTL.dir\EASTL.pdb /FS -c C:\Users\Nick\vcpkg\buildtrees\eastl\src\5ab9f1222a-26ecaf23c0.clean\source\thread_support.cpp
cl : Command line warning D9025 : overriding '/W3' with '/W4'
ninja: build stopped: subcommand failed.
```
-Please attach any additional failure logs mentioned in the console output.

**Additional context**
Add any other context about the problem here, such as what you have already tried to resolve the issue.

**Comment by LilyWangLL** [comment]
This will be fixed in PR #21538.

**Comment by LilyWangLL** [comment]
Thanks for posting this issue. This issue have been fixed by PR #21538. I am closing this issue for now.
